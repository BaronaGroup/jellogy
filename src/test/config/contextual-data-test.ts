import {prepareTestKarhu} from '../test-utils'

describe('correlation-id-test', () => {
    describe('contextual data string', () => {
        it('includes correlation id in text output, if available', prepareTestKarhu({
            outputFormat: 'text',
            getContextualData: () => 'testId'
        })((karhu, output) => {
            karhu.context('cid-test').info('this is a simple test')
            expect(output.tracked).toMatchSnapshot()
        }))

        it('includes correlation id in json output, if available', prepareTestKarhu({
            outputFormat: 'json',
            getContextualData: () => 'testId'
        })((karhu, output) => {
            karhu.context('cid-test').info({key: 'test-value'})
            expect(output.tracked).toMatchSnapshot()
        }))
    })

    describe('contextual data object', () => {
        it('includes correlation id in text output, if available', prepareTestKarhu({
            outputFormat: 'text',
            getContextualData: () => ({testValue: 'myString', anotherTestValue: '123'})
        })((karhu, output) => {
            karhu.context('cid-test').info('this is a simple test')
            expect(output.tracked).toMatchSnapshot()
        }))

        it('includes correlation id in json output, if available', prepareTestKarhu({
            outputFormat: 'json',
            getContextualData: () => ({testValue: 'alpha', another: 'beta'})
        })((karhu, output) => {
            karhu.context('cid-test').info({key: 'test-value'})
            expect(output.tracked).toMatchSnapshot()
        }))
    })
})
