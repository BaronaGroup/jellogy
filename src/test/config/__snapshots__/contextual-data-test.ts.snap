// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`correlation-id-test contextual data object includes correlation id in json output, if available 1`] = `
Map {
  "NONE" => Array [],
  "DEBUG" => Array [],
  "INFO" => Array [
    Array [
      "{\\"timestamp\\":1,\\"context\\":\\"cid-test\\",\\"logLevel\\":\\"INFO\\",\\"details\\":[{\\"key\\":\\"test-value\\"}],\\"testValue\\":\\"alpha\\",\\"another\\":\\"beta\\"}",
      "INFO",
      "cid-test",
    ],
  ],
  "WARN" => Array [],
  "ERROR" => Array [],
}
`;

exports[`correlation-id-test contextual data object includes correlation id in text output, if available 1`] = `
Map {
  "NONE" => Array [],
  "DEBUG" => Array [],
  "INFO" => Array [
    Array [
      "[1] INFO cid-test this is a simple test { testValue: 'myString', anotherTestValue: '123' }",
      "INFO",
      "cid-test",
    ],
  ],
  "WARN" => Array [],
  "ERROR" => Array [],
}
`;

exports[`correlation-id-test contextual data string includes correlation id in json output, if available 1`] = `
Map {
  "NONE" => Array [],
  "DEBUG" => Array [],
  "INFO" => Array [
    Array [
      "{\\"timestamp\\":1,\\"context\\":\\"cid-test\\",\\"logLevel\\":\\"INFO\\",\\"details\\":[{\\"key\\":\\"test-value\\"}],\\"contextualData\\":\\"testId\\"}",
      "INFO",
      "cid-test",
    ],
  ],
  "WARN" => Array [],
  "ERROR" => Array [],
}
`;

exports[`correlation-id-test contextual data string includes correlation id in text output, if available 1`] = `
Map {
  "NONE" => Array [],
  "DEBUG" => Array [],
  "INFO" => Array [
    Array [
      "[1] INFO cid-test this is a simple test testId",
      "INFO",
      "cid-test",
    ],
  ],
  "WARN" => Array [],
  "ERROR" => Array [],
}
`;
